pokedex-with-bit/app.py

```diff
@@ -25,20 +24,54 @@ def config():
 
 
 def index(num_docs: int):
-    f = Flow.load_config('flows/index.yml')
-    num_docs = min(num_docs, len(glob(os.path.join(os.getcwd(), IMAGE_SRC), recursive=True)))
-    with f:
-        with TimeContext(f'QPS: indexing {num_docs}', logger=f.logger):
-            f.index_files(IMAGE_SRC, request_size=64, read_mode='rb', size=num_docs)
+    num_docs = min(num_docs, len(glob(os.path.join(os.getcwd(), IMAGE_SRC),
+                                      recursive=True)))
 
+    flow = Flow(workspace="workspace")\
+        .add(uses={"jtype": "ImageCrafter",
+                   "with": {"target_size": 96,
+                            "img_mean": [0.485, 0.456, 0.406],
+                            "img_std": [0.229, 0.224, 0.225]}})
+    flow = flow.add(uses=BigTransferEncoder)
+    flow = flow.add(uses={"jtype": "EmbeddingIndexer",
+                          "with": {"index_file_name": "image.json"},
+                          "metas": {"name": "vec_idx"}},
+              name="vec_idx")
+    flow = flow.add(uses={"jtype": "KeyValueIndexer",
+                          "metas": {"name": "doc_idx"}},
+                    name="kv_idx",
+                    needs="gateway")    # to enable parallel running
+    flow = flow.add(name="join_all",
+                    needs=["kv_idx", "vec_idx"],
+                    read_only="true")
 
-def query_restful():
-    f = Flow.load_config('flows/query.yml')
-    f.use_rest_gateway()
+    with flow:
+        flow.index(inputs=DocumentArray.from_files(IMAGE_SRC, size=num_docs),
```

Comment:
General question: is it recommended to use this API in 2.0:  `flow.post('/search', ...)`

Reply:
Maybe you can reuse some of the configs for the query Flow.

Reply:
I think it makes it a bit more explicit what `@requests(on='/search')` is doing so I would favour the call to `Flow.post`

Reply:
@FionnD What do you think?
I do like both actually. f.index() is more readable while f.post(on='/search') is more explicit.

Reply:
I will let @alexcg1 jump in here! He has good views on this stuff

Reply:
I prefer `flow.post`. Explicit is the way to go. If we need to explain we can add a comment

Reply:
https://github.com/jina-ai/examples/pull/626/commits/473cafec51508030ab3c81c8bfa4cc08439a92ec